Code/Engine/Foundation/Types/Implementation/VariantTypeRegistry.cpp

```diff
@@ -0,0 +1,74 @@
+#include <FoundationPCH.h>
+
+#include <Foundation/Reflection/Implementation/RTTI.h>
+#include <Foundation/Types/VariantTypeRegistry.h>
+
+EZ_IMPLEMENT_SINGLETON(ezVariantTypeRegistry);
+
+// clang-format off
+EZ_BEGIN_SUBSYSTEM_DECLARATION(Foundation, VariantTypeRegistry)
+
+BEGIN_SUBSYSTEM_DEPENDENCIES
+"Reflection"
+END_SUBSYSTEM_DEPENDENCIES
+
+ON_CORESYSTEMS_STARTUP
+{
+  EZ_DEFAULT_NEW(ezVariantTypeRegistry);
+}
+
+ON_CORESYSTEMS_SHUTDOWN
+{
+  ezVariantTypeRegistry* pDummy = ezVariantTypeRegistry::GetSingleton();
+  EZ_DEFAULT_DELETE(pDummy);
+}
+
+EZ_END_SUBSYSTEM_DECLARATION;
+// clang-format on
+
+ezVariantTypeRegistry::ezVariantTypeRegistry()
+  : m_SingletonRegistrar(this)
+{
+  ezPlugin::s_PluginEvents.AddEventHandler(ezMakeDelegate(&ezVariantTypeRegistry::PluginEventHandler, this));
+
+  UpdateTypes();
+}
+
+ezVariantTypeRegistry::~ezVariantTypeRegistry()
+{
+  ezPlugin::s_PluginEvents.RemoveEventHandler(ezMakeDelegate(&ezVariantTypeRegistry::PluginEventHandler, this));
+}
+
+const ezVariantTypeInfo* ezVariantTypeRegistry::FindVariantTypeInfo(const ezRTTI* pType) const
+{
+  const ezVariantTypeInfo* pTypeInfo = nullptr;
+  m_TypeInfos.TryGetValue(pType, pTypeInfo);
+  return pTypeInfo;
+}
+
+void ezVariantTypeRegistry::PluginEventHandler(const ezPluginEvent& EventData)
+{
+  UpdateTypes();
```

Comment:
I think you could filter out a few redundant events.

Reply:
I hope this us correct now. I look at AfterLoadingBeforeInit  and AfterUnloading